learn/shorts/intro-gh-actions.qmd

```diff
@@ -0,0 +1,369 @@
+``` yaml
+jobs:
+  job-name:
+    # Build on ubuntu, which is Linux.
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository by cloning it.
+        # A common action to use, available from GitHub.
+        uses: actions/checkout@v2
+```
+
+You can keep chaining steps together, as many as you need, to complete
+the task you want your action to do.
+
+### Creating a simple GitHub Action
+
+Imagine you want to create a simple website, using
+[Quarto](https://quarto.org) (which we use to build this website) to
+build a website for you and publish it to [GitHub
+Pages](https://pages.github.com/) so it can be hosted online and be
+available to anyone. Like many tools, Quarto provides a
+[template](https://github.com/quarto-dev/quarto-actions/blob/main/examples/quarto-publish-example.yml)
+action file for you to use.
+
+What you would like is to have GitHub take the Quarto `.qmd` files,
+render them into HTML, and then publish them to GitHub Pages as a
+website. At least for this learning short, we wouldn't get into any
+details on using Quarto, we'll only add the GitHub Action to the
+repository that will build a website for you using Quarto.
+
+1.  Create a new GitHub repository, call it anything you like. Something
+    simple like `simple-website`.
+2.  Clone the repository to your computer.
+3.  Create a new file in the `.github/workflows/` folder called
+    `build-website.yml`.
+4.  Open the file in a text editor and start writing out these "code"
+    instructions.
+
+We'll start with the `on` section, as well as the `name` of the Action.
+In this case, we want the Action to always run whenever you or someone
+else pushes changes to the `main` branch of the repository. So we would
+start with:
+
+``` yaml
+name: Build website
+on:
+  push:
+    branches:
+      - main
+```
+
+Because we want to GitHub to host the website on their GitHub Pages
+service, which uses a special branch called `gh-pages` in the
+repository, we need to give the Action some permissions to be able to
+write to that branch. We'll add that by using the `permissions` section:
+
+``` yaml
+permissions: 
+    contents: write
+    pages: write
+```
+
+Next is the core part of the Action, the `jobs` section. Because we want
+to build the contents of the repository using Quarto and then push to
+the `gh-pages` branch, we need at least three steps:
+
+1.  Downloading the contents of the repository into the GitHub server by
+    checking out the repository.
+2.  Installing Quarto in the server.
+3.  Running Quarto on the repository contents to build the website and
+    store the output in the branch called `gh-pages`.
+
+Thankfully, we don't have to do all of these steps manually, since there
+already exists Actions that you can download and use, such as the
+`checkout` action provided by GitHub's `actions` account, along with the
+`quarto-actions/setup` and `quarto-actions/publish` actions provided by
+the Quarto team in their `quarto-dev` GitHub account.
```

Comment:
Can these be found in GitHub's marketplace? If yes, maybe add that as a source instead to keep things simple?

Reply:
They are not found in the marketplace, only from their repository.